File: 36-valid-sudoku/valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
       vector<set<char>> rows(9);
       vector<set<char>> cols(9);
       vector<set<char>> sub(9);

        for(int i=0; i<board.size(); i++) {
            for(int j=0; j<board[0].size(); j++) {
                char cur = board[i][j];
                if(cur == '.') continue;
                int subIndex = 3 * (i / 3) + j / 3;

                if(rows[i].find(cur) != rows[i].end() || 
                   cols[j].find(cur) != cols[j].end() ||
                   sub[subIndex].find(cur) != sub[subIndex].end()) {

                    return false;
                }
                rows[i].insert(cur);
                cols[j].insert(cur);
                sub[subIndex].insert(cur);
            }
        }
        return true;
    }
};

File: 36-valid-sudoku/valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        rows = {i:set() for i in range(10)}
        cols = {i: set() for i in range(10)}
        sub = {i: set() for i in range(10)}

        m, n = len(board), len(board[0])
        for i in range(m):
            for j in range(n):
                if board[i][j] == '.':
                    continue
                subIndex = (3 * (j // 3)) + (i // 3)
                print(subIndex)
                if board[i][j] in rows[i] or board[i][j] in cols[j] or board[i][j] in sub[subIndex]:
                    return False
                rows[i].add(board[i][j])
                cols[j].add(board[i][j])
                sub[subIndex].add(board[i][j])
        return True
                
